composites/Plugin/ContentAnalysis/components/AnalysisCollapsible.js

```diff
@@ -155,23 +132,22 @@ export class AnalysisCollapsible extends React.Component {
 	 * @returns {void}
 	 */
 	toggleOpen() {
+		const { isOpen } = this.state;
+
 		this.setState( {
-			isOpen: ! this.isOpen(),
+			isOpen: ! isOpen,
 		} );
 	}
 
 	/**
-	 * Returns whether or not the collapsible should be rendered open or closed.
+	 * Creates the header by wrapping the IconsButton with a header.
 	 *
-	 * @returns {boolean} Whether or not this component is open.
+	 * @param {object} props The properties for the component.
+	 *
+	 * @returns {ReactElement} The header to render.
 	 */
-	isOpen() {
-		// When `isOpen` is null then the user has never opened or closed the collapsible.
-		if ( this.state.isOpen === null ) {
-			return this.props.initialIsOpen;
-		}
-
-		return this.state.isOpen === true;
+	static getHeading( props ) {
+		return wrapInHeading( StyledIconsButton, props.headingLevel );
```

Comment:
This function appears in this form both in 'AnalysisCollapsible' and 'Collapsible'. Not that problematic, but since it is a static function, I actually suggest calling 'wrapInHeading()' directly in the constructor and whereever else needed.

Reply:
I made it static because it doesn't use any instance variables. And I made it a function because it gets used twice. What is the gain of not extraction this to a function?